lib/releases.js

```diff
@@ -29,14 +32,15 @@ class Release {
  */
 async function fetchReleases () {
   const url = 'https://api.github.com/repos/opentofu/opentofu/releases';
+  const githubToken = core.getInput('github_token');
```

Comment:
Maybe something like this could work?
```suggestion
  const githubToken = core.getInput('github_token');
  if (githubToken == "") {
      githubToken = process.env.GITHUB_TOKEN
  }
```

Reply:
By default, I think it's a good idea to have the setup automatically work from `GITHUB_TOKEN`.

However, if `GITEA_ACTIONS=True` or `FORGEJO_ACTIONS=True` (not yet used) is available, then `GITHUB_TOKEN` should be ignored because it is overridden by Forgejo/Gitea.

Reply:
Here's a list of environmental variables (from Forgejo v1.21.4-0):

```bash
declare -x ACT="true"
declare -x ACTIONS_CACHE_URL="https://***/"
declare -x ACTIONS_RUNTIME_TOKEN="***"
declare -x ACTIONS_RUNTIME_URL="https://***/api/actions_pipeline/"
declare -x AGENT_TOOLSDIRECTORY="/opt/hostedtoolcache"
declare -x CI="true"
declare -x DEBIAN_FRONTEND="noninteractive"
declare -x DEPLOYMENT_BASEPATH="/opt/runner"
declare -x GITEA_ACTIONS="true"
declare -x GITEA_ACTIONS_RUNNER_VERSION="v3.3.0"
declare -x GITHUB_ACTION="1"
declare -x GITHUB_ACTIONS="true"
declare -x GITHUB_ACTION_PATH=""
declare -x GITHUB_ACTION_REF=""
declare -x GITHUB_ACTION_REPOSITORY=""
declare -x GITHUB_ACTOR="bencurio"
declare -x GITHUB_API_URL="https://***/api/v1" # Forgejo Instance
declare -x GITHUB_BASE_REF=""
declare -x GITHUB_ENV="/var/run/act/workflow/envs.txt"
declare -x GITHUB_EVENT_NAME="push"
declare -x GITHUB_EVENT_PATH="/var/run/act/workflow/event.json"
declare -x GITHUB_GRAPHQL_URL=""
declare -x GITHUB_HEAD_REF=""
declare -x GITHUB_JOB="test-setup-opentofu"
declare -x GITHUB_OUTPUT="/var/run/act/workflow/outputcmd.txt"
declare -x GITHUB_PATH="/var/run/act/workflow/pathcmd.txt"
declare -x GITHUB_REF="refs/heads/main"
declare -x GITHUB_REF_NAME="main"
declare -x GITHUB_REF_TYPE="branch"
declare -x GITHUB_REPOSITORY="bencurio/ci-tests"
declare -x GITHUB_REPOSITORY_OWNER="bencurio"
declare -x GITHUB_RETENTION_DAYS=""
declare -x GITHUB_RUN_ID="1315"
declare -x GITHUB_RUN_NUMBER="68"
declare -x GITHUB_SERVER_URL="https://***" # Forgejo Instance
declare -x GITHUB_SHA="cb496311db324d9762081e1d73ae58333eada5f5"
declare -x GITHUB_STATE="/var/run/act/workflow/statecmd.txt"
declare -x GITHUB_STEP_SUMMARY="/var/run/act/workflow/SUMMARY.md"
declare -x GITHUB_TOKEN="***" # Overrides Forgejo/Gitea for compatibility reasons
declare -x GITHUB_WORKFLOW="Test"
declare -x GITHUB_WORKSPACE="/workspace/bencurio/ci-tests"
declare -x HOME="/root"
declare -x HOSTNAME="79c4a49b2c38"
declare -x IMAGE_OS="ubuntu22"
declare -x ImageOS="ubuntu22"
declare -x JOB_CONTAINER_NAME="GITEA-ACTIONS-TASK-2092_WORKFLOW-Test_JOB-x"
declare -x LANG="C.UTF-8"
declare -x LSB_OS_VERSION="2204"
declare -x LSB_RELEASE="22.04"
declare -x OLDPWD
declare -x PATH="/opt/hostedtoolcache/node/18.19.0/x64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
declare -x PWD="/workspace/bencurio/ci-tests"
declare -x RUNNER_ARCH="X64"
declare -x RUNNER_OS="Linux"
declare -x RUNNER_PERFLOG="/dev/null"
declare -x RUNNER_TEMP="/tmp"
declare -x RUNNER_TOOL_CACHE="/opt/hostedtoolcache"
declare -x RUNNER_TRACKING_ID=""
declare -x RUNNER_USER="root"
declare -x RUN_TOOL_CACHE="/opt/hostedtoolcache"
declare -x SHLVL="1"
declare -x USER="root"
```

Reply:
Maybe something this then?
```suggestion
  const githubToken = core.getInput('github_token');
  const isForgejoRunner = process.env.FORGEJO_ACTIONS || process.env.GITEA_ACTIONS;
  if (githubToken == "" && !isForgejoRunner) {
      githubToken = process.env.GITHUB_TOKEN
  }
```

Reply:
Sorry I had a exhausting week so I didn't found time to have a look on it earlier. I'll do some test this week-end an let you know.

Reply:
Hey, no worries @nicolas-brousse thanks for letting us know. Looking forward to your feedback!

Reply:
I did some change here and I tested with my Gitea instance and it seems to work with the change.